File: solutions/javascript/triangle/1/triangle.js
export class Triangle {
  constructor(...sides) {
    this.sides = sides;
  }

  isValid() {
    const [a, b, c] = this.sides;
    // All sides must be positive and satisfy triangle inequality
    return a > 0 && b > 0 && c > 0 &&
           a + b > c &&
           a + c > b &&
           b + c > a;
  }

  get isEquilateral() {
    if (!this.isValid()) return false;
    const [a, b, c] = this.sides;
    return a === b && b === c;
  }

  get isIsosceles() {
    if (!this.isValid()) return false;
    const [a, b, c] = this.sides;
    return a === b || a === c || b === c;
  }

  get isScalene() {
    if (!this.isValid()) return false;
    const [a, b, c] = this.sides;
    return a !== b && a !== c && b !== c;
  }
}

File: solutions/python/triangle/4/triangle.py
"""Evaluate the type of triangle"""


def is_triangle(sides):
    """Check if three sides can form a valid triangle."""
    
    a_Side, b_Side, c_Side = sides
    # All sides must be positive
    if a_Side <= 0 or b_Side <= 0 or c_Side <= 0:
        return False
    # Triangle inequality theorem
    return (a_Side + b_Side >= c_Side) and (b_Side + c_Side >= a_Side) and (a_Side + c_Side >= b_Side)


def equilateral(sides):
    """Check if triangle is equilateral (all sides equal)."""
    
    if not is_triangle(sides):
        return False
    a_Side, b_Side, c_Side = sides
    return a_Side == b_Side == c_Side


def isosceles(sides):
    """Check if triangle is isosceles (at least two sides equal)."""
    
    if not is_triangle(sides):
        return False
    a_Side, b_Side, c_Side = sides
    return a_Side == b_Side or b_Side == c_Side or a_Side == c_Side


def scalene(sides):
    """Check if triangle is scalene (all sides different)."""
    
    if not is_triangle(sides):
        return False
    a_Side, b_Side, c_Side = sides
    return a_Side != b_Side and b_Side != c_Side and a_Side != c_Side